### PR TITLE
Drop Python 2

### LINKED ISSUE #219: Drop Python 2 support
I'd like to propose we drop support for Python 2. This has already been done in Zarr, Dask, and many other projects throughout the community. Are there any objections to this?

### PR DESCRIPTION
**Pull request by jrbourbeau**
Closes #219

TODO:
* [ ] Unit tests and/or doctests in docstrings
* [ ] ``tox -e py38`` passes locally
* [ ] ~~``tox -e py27`` passes locally~~
* [ ] ~~Docstrings and API docs for any new/modified user-facing classes and functions~~
* [x] Changes documented in docs/release.rst
* [ ] ``tox -e docs`` passes locally
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage to 100% (Coveralls passes)

### CODE REVIEW

**Comment by jakirkham** [comment]
FWIW I'm ok skipping the macOS bit and saving that for another PR.

**Comment by jrbourbeau** [comment]
Sounds good! I've skipped the OSX build in this PR and have switched to using GitHub Actions over in https://github.com/zarr-developers/numcodecs/pull/221

**Comment by jrbourbeau** [comment]
Thanks for reviewing @alimanfoo! I've updated the docstring for `Buffer` to indicate it's just a convenience class and also removed some lingering Python 2 references in documentation and our `tox` setup

File: numcodecs/compat_ext.pyx [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
 from cpython.buffer cimport PyObject_GetBuffer, PyBuffer_Release
 
 
 from .compat import ensure_contiguous_ndarray
 
 
 cdef class Buffer:
-    """Compatibility class to work around fact that array.array does not support
-    new-style buffer interface in PY2."""
+    """Convenience class for buffer interface"""
```
Comment (alimanfoo) [inline numcodecs/compat_ext.pyx:13]:
```suggestion
    """Convenience class for buffer interface."""
```

Thanks for updating this docstring. Apologies, I am a total obsessive when it comes to punctuation :-)

Comment (jrbourbeau) [reply numcodecs/compat_ext.pyx]: No problem : )

**Comment by jakirkham** [comment]
Thanks James for the PR and Alistair for the review! 😄

**Comment by joshmoore** [comment]
Looks like readthedocs is going to need updating to use python 3 as well: https://readthedocs.org/projects/numcodecs/builds/11416597/

**Comment by jakirkham** [comment]
Thanks Josh 😄

**Pull request merged**